pytrip/vdx.py

```diff
@@ -1026,17 +1032,37 @@ def read_dicom(self, info, data):
             _z_pos = contour.ContourData[2]
 
             # if we have a new z_position, add a new slice object to self.slices
-            if _z_pos not in [i.get_position() for i in self.slices]:
+            if _z_pos not in [_slice.get_position() for _slice in self.slices]:
                 logger.debug("Append new slice at z_position {:f} to slices list:".format(_z_pos))
                 sl = Slice(cube=self.cube)
-                sl.add_dicom_contour(contour)
-                # TODO: check on con.ContourGeometricType = 'CLOSED_PLANAR'
-                # so far we simply assume all contours from DICOM are closed.
-                if sl.contour[-1].number_of_points() > 3:
-                    sl.contour[-1].contour_closed = True
-                else:
-                    sl.contour[-1].contour_closed = False
                 self.slices.append(sl)
+            else:
+                # lookup proper slice (just to be sure, should the contours come in random order)
```

Comment:
@grzanka At this point in the code a marker could be added, this part of code is only executed if there are multiple contours for the same VOI in the same slice.

Then dicom2trip.py can be used to find datasets in slicerRT db which could be suitable for testing.